### PR TITLE
Add comms handler

### PR DESCRIPTION
**Pull request by BrianWoodruff**
Please review carefully.  There are many places I could have introduced bugs.

### CODE REVIEW

File: ovve_ui/display/widgets.py [hunk 1]
```diff
@@ -22,19 +22,19 @@ def initializeHomeScreenWidget(window: MainWindow) -> None:
...
+        window.get_mode_display(window.settings.mode))
     window.mode_button_main.clicked.connect(lambda: window.display(1))
 
     window.resp_rate_button_main = window.makeFancyDisplayButton(
         "Resp. Rate", window.settings.resp_rate, "b/min")
     window.resp_rate_button_main.clicked.connect(lambda: window.display(2))
 
     window.minute_vol_button_main = window.makeFancyDisplayButton(
-        "Minute Volume", window.settings.minute_volume, "l/min")
+        "Minute Volume", window.settings.tv, "l/min")
```
Comment (avigupta33) [inline ovve_ui/display/widgets.py:33]: Why was minute_volume changed to tv here?

Comment (avigupta33) [reply ovve_ui/display/widgets.py:33]: Never mind, I think I understand. You are just renaming, right?

File: ovve_ui/ovve_ui.py [hunk 2]
```diff
@@ -23,29 +22,30 @@
...
                              initializeIERatioWidget)
 from utils.params import Params
 from utils.settings import Settings
-
+from utils.comms_adapter import CommsAdapter
+from utils.comms_simulator import CommsSimulator
 
 class MainWindow(QWidget):
-    def __init__(self, debug: bool = True) -> None:
+    def __init__(self) -> None:
```
Comment (avigupta33) [inline ovve_ui/ovve_ui.py:37]: Why did you remove the debug mode?

File: ovve_ui/ovve_ui.py [hunk 3]
```diff
@@ -118,26 +135,28 @@ def initalizeAndAddStackWidgets(self) -> None:
     def display(self, i: int) -> None:
         self.stack.setCurrentIndex(i)
 
-    def update_main_page_ui(self) -> None:
+    def update_ui(self, params: Params) -> None:
+        self.params = params
```
Comment (avigupta33) [inline ovve_ui/ovve_ui.py:162]: This will possibly cause a bug where which will prevent the user from changing settings. It is no problem though, I can fix this later on after I redesign how pages work.

File: ovve_ui/ovve_ui.py [hunk 4]
```diff
@@ -148,81 +167,47 @@ def updateGraphs(self) -> None:
...
         self.resp_rate_page_rect.updateValue(self.local_settings.resp_rate)
 
     def decrementRespRate(self) -> None:
-        self.local_settings.resp_rate -= self.settings.resp_rate_increment
+        self.local_settings.resp_rate -= self.resp_rate_increment
         self.resp_rate_page_rect.updateValue(self.local_settings.resp_rate)
 
     def incrementMinuteVol(self) -> None:
-        self.local_settings.minute_volume += self.settings.minute_volume_increment
+        self.local_settings.tv += self.minute_volume_increment
```
Comment (avigupta33) [inline ovve_ui/ovve_ui.py]: Do we want to rename minute_volume_increment to be more consistent?

File: ovve_ui/ovve_ui.py [hunk 5]
```diff
@@ -235,64 +220,81 @@ def commitRespRate(self) -> None:
...
                 "Minute Vol",
-                self.settings.minute_volume,
-                self.local_settings.minute_volume,
+                self.settings.tv,
+                self.local_settings.tv,
             ))
-        self.settings.minute_volume = self.local_settings.minute_volume
-        self.minute_vol_button_main.updateValue(self.settings.minute_volume)
+        self.settings.tv = self.local_settings.tv
+        self.minute_vol_button_main.updateValue(self.settings.tv)
```
Comment (avigupta33) [inline ovve_ui/ovve_ui.py:257]: Can we rename minute_vol_button_main to tv_button_main?

File: ovve_ui/display/widgets.py (hunk 1 above)
Comment (BrianWoodruff) [reply ovve_ui/display/widgets.py:33]: Yes, I'm renaming to be consistent with the names in the incoming packets

File: ovve_ui/ovve_ui.py (hunk 2 above)
Comment (BrianWoodruff) [reply ovve_ui/ovve_ui.py:37]: I think debug can be handled through the simulator.  All debug mode did is set some initial values.  I thought those functions didn't belong in the params or settings classes

File: ovve_ui/ovve_ui.py (hunk 3 above)
Comment (BrianWoodruff) [reply ovve_ui/ovve_ui.py:162]: Yes, I figured this would be addressed with the redesign

File: ovve_ui/ovve_ui.py (hunk 4 above)
Comment (BrianWoodruff) [reply ovve_ui/ovve_ui.py]: Yes, I think so.  Also, you'll see I moved the increments to the MainWindow class.  I'm not sure they belong there but I think they are a property of the UI, not the incoming data so I removed them from settings.

File: ovve_ui/ovve_ui.py (hunk 5 above)
Comment (BrianWoodruff) [reply ovve_ui/ovve_ui.py:257]: yes

File: ovve_ui/ovve_ui.py (hunk 3 above)
Comment (avigupta33) [reply ovve_ui/ovve_ui.py:162]: Yes it will!

**Comment by zmk5** [comment]
Looking through it now!

File: ovve_ui/utils/comms_simulator.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        params_dict = json.loads(params_str)
+
+        while True:
+            params_dict["peep"] = random.randrange(3, 6)
+            params_dict["tv_insp"] = random.randrange(475, 575)
+            params_dict["tv_exp"] = random.randrange(475, 575)
+            params_dict["ppeak"] = random.randrange(15, 20)
+            params_dict["pplat"] = random.randrange(15, 20)
+            self.comms_adapter.update_params(params_dict)
+            time.sleep(1)
```
Comment (zmk5) [inline ovve_ui/utils/comms_simulator.py:55]: One thing I think might be better is to avoid using sleep maybe. If I remember correctly, it can create issues with PyQt5, so we may want to use their version which is `QTimer` (it is a module in `QtCore`).

File: ovve_ui/utils/params.py [hunk 7]
```diff
@@ -5,34 +5,76 @@
...
 
-    def from_dict(self, params_dict: dict) -> None:
-        self.peep = params_dict['peep']
-        self.tv_insp = params_dict['tv_insp']
-        self.tv_exp = params_dict['tv_exp']
-        self.ppeak = params_dict['ppeak']
-        self.pplat = params_dict['pplat']
-        
\ No newline at end of file
+    def from_dict(self, j: dict) -> None:
```
Comment (zmk5) [inline ovve_ui/utils/params.py:59]: We may want to add a means of error handling with this, but it might involve making an attribute creating the keys we are actually looking for and returning `None` if a name isn't coming out right.

Comment (BrianWoodruff) [reply ovve_ui/utils/params.py:59]: I agree - populating dictionaries can be error prone.  I'm not sure how to implement what you mean.  Can you add it in my branch?

Comment (zmk5) [reply ovve_ui/utils/params.py:59]: Yeah I can try and figure it out and add something.

Comment (BrianWoodruff) [reply ovve_ui/utils/params.py:59]: I put a todo tag on both dictionary conversions.  Let's address it later.
